src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -58,9 +57,20 @@ public class BindingsGenerator : AndroidRunToolTask
 		protected override string DefaultErrorCode => "BG0000";
 
 		private List<Tuple<string, string>> transform_files = new List<Tuple<string,string>> ();
+		bool useDotNet = false;
+		string generatorAssemblyPath;
 
 		public override bool RunTask ()
 		{
+			var framework = RuntimeInformation.FrameworkDescription;
+			Log.LogDebugMessage ($"FrameworkDescription: {framework}");
+			if (!framework.StartsWith (".NET Framework", StringComparison.OrdinalIgnoreCase) &&
+					!framework.StartsWith ("Mono", StringComparison.OrdinalIgnoreCase)) {
```

Comment:
This is the only part I'm "iffy" about. Is there any better way to detect we're running on .NET 5/.NET Core from a netstandard library?

I got this idea from: https://github.com/dotnet/BenchmarkDotNet/blob/a08062fa473e3426c7d5f1a1cf2ad5cbbc46f7d1/src/BenchmarkDotNet/Environments/Runtimes/CoreRtRuntime.cs#L47-L51

Reply:
Related to the "iffy-ness", should we "abstract-ify" this a bit, so that it's not *just* `generator.exe`/`generator.dll` which is special-cased?  From a quick `git grep -3 'override.*ToolName' src/Xamarin.Android.Build.Tasks`, these other utilities may also need to be similarly special-cased:

  * `javadoc-to-mdoc.exe` (`<ImportJavaDoc/>`)
  * `mdoc.exe` (`<MDoc/>`)
  * `mono-symbolicate.exe` (?) (`<MonoSymbolicate/>`)

(xamarin-android builds `javadoc-to-mdoc.exe`, and thus *can* add .NET Core target framework build support for that.  `mdoc.exe` and `mono-symbolicate.exe` come from Mono, and thus need some cross-team discussion to determine how to deal with them in the future.  *Presumably* they'll become .NET Core apps?)

Expanding our scope thus, how can/should these be supported?  For starters, we'd want `AndroidRunToolTask` to have the appropriate logic, not `BindingsGenerator`/`ImportJavaDoc`/etc.

Next, do we *need* care about ".NET Core" vs. everything else?  Or can we just *probe*, e.g. by having `GenerateCommandLineCommands()` do:

```csharp
string dotnetCommand = Path.ChangeExtension (ToolName, ".dll");
string dotnetCommandPath = Path.Combine (ToolPath, dotnetCommand);
if (File.Exists (dotnetCommandPath)) {
    // use `dotnet {dotnetCommandPath} …`
}
else {
    // use `{ToolPath}/{ToolName} …`
}
```

Reply:
MSBuild knows what it is running on if that helps: `$(MSBuildRuntimeType) = Core, Full, Mono`: https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-reserved-and-well-known-properties?view=vs-2019.

Reply:
Thanks, maybe I should switch to it, since MSBuild is defining these based on how it is built:
https://github.com/microsoft/msbuild/blob/6f41a4fa3c6957d430783b7d31fe16e6a13b6e8e/src/Build/Evaluation/Evaluator.cs#L1159-L1166